### PR TITLE
Bump version to 0.23.0

### PR DESCRIPTION
**Pull request by Yadunund**
Bumping minor version so that `rolling` remains ahead of `iron`

### CODE REVIEW

**Review by clalancette — APPROVED** [review]
This looks good to me, but I'd like another review from the rosbag2 maintainers before we merge this.

**Review by MichaelOrlov — APPROVED** [review]

**Pull request merged**
